test/image/mocks/z-new_tickmode_sync.json

```diff
@@ -0,0 +1,87 @@
+{
+   "data": [
+      {
+         "name": "Apples",
+         "type": "bar",
+         "x": [
+            "Jan",
+            "Feb",
+            "Mar",
+            "Apr",
+            "May"
+         ],
+         "y": [
+            232,
+            2506,
+            470,
+            1864,
+            -190
+         ]
+      },
+      {
+         "name": "Oranges",
+         "type": "scatter",
+         "x": [
+            "A",
+            "B",
+            "C"
+        ],
+        "y": [
+            0,
+            0.5,
+            1
+        ],
+         "yaxis": "y2",
+         "xaxis": "x2"
+      }
+   ],
+   "layout": {
+      "margin": {
+         "t": 40,
+         "r": 70,
+         "b": 40,
+         "l": 70
+      },
+      "width": 700,
+      "showlegend": false,
+      "xaxis2": {
+         "anchor": "y2",
+         "side": "top",
+         "overlaying": "x",
+         "tickmode": "sync",
+         "domain": [
+            0.52,
+            1
+         ]
+      },
+      "yaxis": {
+         "title": {
+            "text": "Apples"
+         },
+         "side": "left",
```

Comment:
If I add `"gridcolor": "black",` here I don't see the effect on the major grid lines because `yaxis2` draws grid lines over at the same position.
To fix this issue please make the change so that `axes.drawGrid` function skip drawing of any grid lines for synced axes.